packages/lit-node-client-nodejs/src/lib/lit-node-client-nodejs.ts

```diff
@@ -830,7 +843,7 @@ export class LitNodeClientNodeJs
     };
 
     // -- execute
-    keys.forEach(async (key) => {
+    for await (const key of keys) {
```

Comment:
This surely had to be awaited. Without awaiting `signatures` was returned before it was populated with the result at the end of each iteration

Reply:
Wondering if we just want to make this a normal `for` loop and ditch the iterators.

Reply:
Do we know if somebody was using this? The only way I see this working is that wasm was already loaded and it behaved completely synchronously regardless of the `async` and `await` calls.
If not, then the response would be a `{}`, generating race condition on that and prone to a future update breaking it.

Reply:
The warm calls are currently synchronous operations. In `v7` this will no longer be true, but not because the wasm execution has become asynchronous. Rather we have made the api async as to future proof it in some ways.

I don't think we need the async wrappers, personally would rather this just turn into a `for` loop as the iteration that using `for of` and `forEach` seems like will cause higher GC times.

Reply:
In this loop, [`combineEcdsaShares` is async](https://github.com/LIT-Protocol/js-sdk/blob/3eaa12a99e3f085b62b5d1c3aa1864cd40af4318/packages/lit-node-client-nodejs/src/lib/lit-node-client-nodejs.ts#L929) and internally it calls `ecdsaCombine` and `ecdsaVerify` both of which are defined `async` because of a call to `loadModules`

At least when loading the wasm, they are asynchronous and will release the event loop. We are at v7 here
Or I am missing what you mean?